File: android/src/main/kotlin/au/com/annon/flutter_mapbox_turn_by_turn/models/MapboxEnhancedLocationChangeEvent.kt
package au.com.annon.flutter_mapbox_turn_by_turn.models

class MapboxEnhancedLocationChangeEvent(location: com.mapbox.common.location.Location) {
    val isEnhancedLocationChangeEvent = true
    var latitude: Double? = null
    var longitude: Double? = null

    init {
        latitude = location.latitude
        longitude = location.longitude
    }
}

File: ios/Classes/models/MapboxEnhancedLocationChangeEvent.swift
import Foundation

public class MapboxEnhancedLocationChangeEvent: Codable {
  let isEnhancedLocationChangeEvent: Bool
  let latitude: Double?
  let longitude: Double?

  init(latitude: Double?, longitude: Double?) {
    self.isEnhancedLocationChangeEvent = true
    self.latitude = latitude
    self.longitude = longitude
  }
}
